### PR TITLE
Feature 01 & 02

### PR DESCRIPTION
**Pull request by dinccey**

### CODE REVIEW

File: src/main/java/ibmix/kickstart/bikeshop/data/entities/Bicikl.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.*;
+@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
+@AllArgsConstructor
+@Entity
+@Table(name = "Bicikl", schema = "public", catalog = "ibmix")
+public class Bicikl {
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/data/entities/Bicikl.java:13]: English please :D I'm not Kermek, neither are your future clients.

Comment (jsolja) [reply src/main/java/ibmix/kickstart/bikeshop/data/entities/Bicikl.java:13]: And on all other classes

File: src/main/java/ibmix/kickstart/bikeshop/rest/BiciklController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Autowired
+    BiciklRepository biciklRepository;
+
+    @Autowired
+    EntityManager entityManager;
+
+    @GetMapping("/greeting")
+    public String greeting() {
+        return "Hello World!";
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/rest/BiciklController.java:25]: Remove this, it's not needed.

File: src/main/java/ibmix/kickstart/bikeshop/data/DataSourceConfig.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+@Configuration
+public class DataSourceConfig {
+
+    @Bean
+    public DataSource getDataSource(){
+        DataSourceBuilder dataSourceBuilder = DataSourceBuilder.create();
+        dataSourceBuilder.driverClassName("org.postgresql.Driver");
+        dataSourceBuilder.url("jdbc:postgresql://localhost:5432/ibmix");
+        dataSourceBuilder.username("admin");
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/data/DataSourceConfig.java:17]: DB credentials should be in property file.

Comment (jsolja) [reply src/main/java/ibmix/kickstart/bikeshop/data/DataSourceConfig.java:17]: And other connection configuration.

**Comment by jsolja** [comment]
Only pom.xml and src should be committed to repository. Everything else is your local stuff, especially generated tmp classes. Remove it from PR.

File: src/main/java/ibmix/kickstart/bikeshop/data/entities/Bicikl.java (hunk 1 above)
Comment (dinccey) [reply src/main/java/ibmix/kickstart/bikeshop/data/entities/Bicikl.java:13]: Affirmative
XD

File: src/main/java/ibmix/kickstart/bikeshop/data/repositories/BiciklRepository.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package ibmix.kickstart.bikeshop.data.repositories;
+
+import ibmix.kickstart.bikeshop.data.entities.Bicycle;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface BiciklRepository extends JpaRepository<Bicycle,Long> {
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/data/repositories/BiciklRepository.java:8]: English

File: src/main/java/ibmix/kickstart/bikeshop/rest/BiciklController.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.persistence.EntityManager;
+
+@RestController
+public class BiciklController {
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/rest/BiciklController.java:15]: English

File: src/main/java/ibmix/kickstart/bikeshop/rest/BicycleController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.persistence.EntityManager;
+
+@RestController
+public class BicycleController {
+
+    @Autowired
+    BicycleRepository bicycleRepository;
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/rest/BicycleController.java:17]: private

File: src/main/java/ibmix/kickstart/bikeshop/rest/BicycleController.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    EntityManager entityManager;
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/rest/BicycleController.java:20]: private

File: src/main/java/ibmix/kickstart/bikeshop/rest/BicycleController.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    @PostMapping(value="/bicycles",consumes = "application/json", produces = "application/json")
+    public Bicycle newBicycle(@RequestBody Bicycle bicycle){
```
Comment (jsolja) [inline src/main/java/ibmix/kickstart/bikeshop/rest/BicycleController.java:23]: final

**Review by jsolja — APPROVED** [review]

**Pull request merged**